src/swupd/manifest.go

```diff
@@ -0,0 +1,118 @@
+package swupd
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// ManifestHeader contains metadata for the manifest
+type ManifestHeader struct {
+	Format      int
+	Version     int32
+	Previous    int32
+	FileCount   int32
+	TimeStamp   time.Time
+	ContentSize int64
+	Includes    []Manifest
+}
+
+// Manifest represents a bundle or list of bundles (MoM)
+type Manifest struct {
+	Name         string
+	Header       ManifestHeader
+	Files        []File
+	AddedCount   uint32
+	RemovedCount uint32
+	ChangedCount uint32
+}
+
+// MoM is a manifest of manifests with the same header information
+type MoM struct {
+	Header       ManifestHeader
+	SubManifests []Manifest
+}
+
+// readManifestFileHeaderLine Read a header line from a manifest
+func readManifestFileHeaderLine(fields []string, m *Manifest) error {
+	var err error
+	var parsed uint64
+
+	// Only search for defined fields
+	switch fields[0] {
+	case "MANIFEST":
+		if parsed, err = strconv.ParseUint(fields[1], 10, 16); err != nil {
+			return err
+		}
+		m.Header.Format = int(parsed)
+	case "version:":
+		if parsed, err = strconv.ParseUint(fields[1], 10, 32); err != nil {
+			return err
+		}
+		m.Header.Version = int32(parsed)
+	case "previous:":
+		if parsed, err = strconv.ParseUint(fields[1], 10, 32); err != nil {
+			return err
+		}
+		m.Header.Previous = int32(parsed)
+	case "filecount:":
+		if parsed, err = strconv.ParseUint(fields[1], 10, 32); err != nil {
+			return err
+		}
+		m.Header.FileCount = int32(parsed)
+	case "timestamp:":
+		if parsed, err = strconv.ParseUint(fields[1], 10, 64); err != nil {
+			return err
+		}
+		// parsed is already uint64
+		m.Header.TimeStamp = time.Unix(int64(parsed), 0)
+	case "contentsize:":
+		if parsed, err = strconv.ParseUint(fields[1], 10, 64); err != nil {
+			return err
+		}
+		// parsed is already uint64
+		m.Header.ContentSize = int64(parsed)
```

Comment:
One idea: use ParseInt and verify that the value is not negative. That way you don't need to care about overflow here.